src/views/design/ScriptsOverview/index.tsx

```diff
@@ -312,15 +321,43 @@ const ScriptsOverview = withStyles(styles)(
                     <Box paddingBottom={5} marginX={2.8}>
                         { !hasError && (
                             <GenericList
-                                listActions={[
+                                listActions={[].concat(
                                     {
                                         icon: <PlayArrowRounded />,
                                         label: translator('scripts.overview.list.actions.execute'),
                                         onClick: this.setScriptToExecute,
-                                    }, {
+                                        hideAction: (item: IListItem<IColumnNames>) =>
+                                            !checkAuthority(
+                                                SECURITY_PRIVILEGES.S_EXECUTION_REQUEST_WRITE,
+                                                item.columns.securityGroupName.toString(),
+                                            ),
+                                    },
+                                    {
                                         icon: <Edit />,
                                         label: translator('scripts.overview.list.actions.edit'),
-                                        onClick: (id) => {
+                                        onClick: (id: string) => {
+                                            const scripts = getAsyncScripts(this.props.state);
+                                            const selectedScript = scripts.find((item) =>
+                                                getUniqueIdFromScript(item) === id);
+
+                                            redirectTo({
+                                                routeKey: ROUTE_KEYS.R_SCRIPT_DETAIL,
+                                                params: {
+                                                    name: selectedScript.name,
+                                                    version: selectedScript.version.number,
+                                                },
+                                            });
+                                        },
+                                        hideAction: (item: IListItem<IColumnNames>) =>
+                                            !checkAuthority(
+                                                SECURITY_PRIVILEGES.S_SCRIPTS_WRITE,
+                                                item.columns.securityGroupName.toString(),
+                                            ),
+                                    },
+                                    {
+                                        icon: <Visibility />,
+                                        label: translator('scripts.overview.list.actions.edit'),
+                                        onClick: (id: string) => {
```

Comment:
for the label: translator('scripts.overview.list.actions.edit'), we can replace by something like  translator('scripts.overview.list.actions.view'), and add new line in the yammel file iesi-ui\src\views\translations\en_GB.yml